### PR TITLE
release 2.25.0-rc.0

### PR DESCRIPTION
**Pull request by roidelapluie**

### CODE REVIEW

**Comment by roidelapluie** [comment]
I have reverted the 429 handling becauseI want to figure out additional questions first, like, should we retry when there is no Retry header.

It seems that we need a few more days to have the exemplars ready so let's skip it for this release.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,25 @@
+## 2.25.0-rc.0 / 2021-02-12
+
+* [FEATURE] API/Experimental: Accept remote_write requests. #8424
+* [FEATURE] PromQL/Experimental: Add '@ <timestamp>' selector. #8121 #8436 #8425
+* [ENHANCEMENT] Add optional name property to testgroup for better test failure output. #8440
+* [ENHANCEMENT] Added warnings into React Panel on the Graph page. #8427
```
Comment (SuperQ) [inline CHANGELOG.md:6]:
```suggestion
* [ENHANCEMENT] Add warnings into React Panel on the Graph page. #8427
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+* [ENHANCEMENT] TSDB: Increase the buckets for compaction duration. #8342
```
Comment (SuperQ) [inline CHANGELOG.md:7]:
```suggestion
* [ENHANCEMENT] TSDB: Increase the number of buckets for compaction duration metric. #8342
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+* [ENHANCEMENT] Remote: allow passing along custom remote_write HTTP headers. #8416
```
Comment (SuperQ) [inline CHANGELOG.md:8]:
```suggestion
* [ENHANCEMENT] Remote: Allow passing along custom remote_write HTTP headers. #8416
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,25 @@
...
+* [ENHANCEMENT] Mixins: Scope grafana configuration. #8332
+* [ENHANCEMENT] Kubernetes SD: Add endpoint labels metadata. #8273
+* [ENHANCEMENT] UI: Expose total number of label pairs in head in TSDB stats page. #8343
+* [ENHANCEMENT] TSDB: Reload blocks every minute, to detect new blocks and enforce retention more often. #8343
+* [BUGFIX] API: Fix global URL when external address has no port. #8359
+* [BUGFIX] Backfill: Fix error message handling. #8432
+* [BUGFIX] Deprecate --alertmanager.timeout. That flag was not used. #8407
```
Comment (SuperQ) [inline CHANGELOG.md:15]:
```suggestion
* [BUGFIX] Deprecate unused flag `--alertmanager.timeout`. #8407
```

**Comment by codesome** [comment]
> It seems that we need a few more days to have the exemplars ready so let's skip it for this release.

It is ready to go in today, can we re-consider it? (push the last few nits now)

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,26 @@
+## 2.25.0-rc.0 / 2021-02-12
+
+* [FEATURE] API/Experimental: Accept remote_write requests. #8424
+* [FEATURE] PromQL/Experimental: Add '@ <timestamp>' selector. #8121 #8436 #8425
+* [FEATURE] Experimental: Add exemplars support. #6635
+* [ENHANCEMENT] Add optional name property to testgroup for better test failure output. #8440
+* [ENHANCEMENT] Add warnings into React Panel on the Graph page. #8427
+* [ENHANCEMENT] TSDB: Increase the number of buckets for the  compaction duration metric. #8342
```
Comment (beorn7) [inline CHANGELOG.md:8]: Nit: Spurious blank.

**Review by beorn7 — APPROVED** [review]
Looks good generally.

Should we perhaps mark changes that are behind a feature flag as such?

We also haven't been terribly consistent how to mark experimental changes (`**experimental**`, `[EXPERIMENTAL]`, …). Some typographic emphasis would be good in any case. And then let's establish a consistent visible way of marking experimental changes and changes behind a feature flag (where I guess it is implied that the latter are _also_ experimental, but we could also call that out explicitly).

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,26 @@
...
+* [ENHANCEMENT] Remote: Allow passing along custom remote_write HTTP headers. #8416
+* [ENHANCEMENT] Mixins: Scope grafana configuration. #8332
+* [ENHANCEMENT] Kubernetes SD: Add endpoint labels metadata. #8273
+* [ENHANCEMENT] UI: Expose total number of label pairs in head in TSDB stats page. #8343
+* [ENHANCEMENT] TSDB: Reload blocks every minute, to detect new blocks and enforce retention more often. #8343
+* [BUGFIX] API: Fix global URL when external address has no port. #8359
+* [BUGFIX] Backfill: Fix error message handling. #8432
+* [BUGFIX] Deprecate unused flag --alertmanager.timeout. #8407
+* [BUGFIX] Remote: Adapt metric name in remote-write alert rules mixin. #8423
```
Comment (csmarchbanks) [inline CHANGELOG.md:17]: This should probably be labeled as `Mixins` as there were no changes to the Remote code for it. Also, it is pretty vague, perhaps `Mixins: Support remote-write metrics renamed in v2.23 in alerts`?

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,26 @@
+## 2.25.0-rc.0 / 2021-02-12
+
+* [FEATURE] API/Experimental: Accept remote_write requests. #8424
+* [FEATURE] PromQL/Experimental: Add '@ <timestamp>' selector. #8121 #8436 #8425
+* [FEATURE] Experimental: Add exemplars support. #6635
```
Comment (codesome) [inline CHANGELOG.md:5]: https://github.com/prometheus/prometheus/pull/6635#issuecomment-778206405

**Review by csmarchbanks — COMMENTED** [review]
Generally LGTM, with one small suggestion. I think reverting the 429 change makes a lot of sense. 

Should the revert happen in a separate PR merged before this? As long as there ends up being a commit with just the revert I am fine either way.

**Comment by roidelapluie** [comment]
PTAL, I have addressed the comments.

**Review by csmarchbanks — APPROVED** [review]
:+1:

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,28 @@
+## 2.25.0-rc.0 / 2021-02-12
+
+This release includes a new `--enable-feature=` flag that enables
+experimental features. Such features might be changed or removed in the future.
+
+* [FEATURE] **experimental** API: Accept remote_write requests. Behind the --enable-feature=remote-write-receiver flag. #8424
+* [FEATURE] **experimental** PromQL: Add '@ <timestamp>' selector. Behind the --enable-feature=promql-at-modifier flag. #8121 #8436 #8425
```
Comment (beorn7) [inline CHANGELOG.md:7]: To be super strict: `@` is a modifier, not a selector.

**Comment by beorn7** [comment]
To clarify: Exemplar support has _not_ made it in here.

**Review by beorn7 — APPROVED** [review]
Approving again, in case my previous one is lacking credibility after all my comments.

**Pull request merged**